src/Drupal/Commands/core/MigrateRunnerCommands.php

```diff
@@ -0,0 +1,706 @@
+<?php
+
+namespace Drush\Drupal\Commands\core;
+
+use Consolidation\OutputFormatters\StructuredData\RowsOfFields;
+use Drupal\Core\Datetime\DateFormatter;
+use Drupal\Core\KeyValueStore\KeyValueFactoryInterface;
+use Drupal\migrate\Exception\RequirementsException;
+use Drupal\migrate\MigrateMessageInterface;
+use Drupal\migrate\Plugin\MigrationInterface;
+use Drupal\migrate\Plugin\MigrationPluginManagerInterface;
+use Drupal\migrate\Plugin\RequirementsInterface;
+use Drush\Commands\DrushCommands;
+use Drush\Drupal\Migrate\MigrateCommandProgressBar;
+use Drush\Drupal\Migrate\MigrateExecutable;
+use Drush\Drupal\Migrate\MigrateMessage;
+use Drush\Drupal\Migrate\MigrateUtils;
+use Webmozart\PathUtil\Path;
+
+class MigrateRunnerCommands extends DrushCommands
+{
+    /**
+     * Migration plugin manager service.
+     *
+     * @var \Drupal\migrate\Plugin\MigrationPluginManagerInterface
+     */
+    protected $migrationPluginManager;
+
+    /**
+     * Date formatter service.
+     *
+     * @var \Drupal\Core\Datetime\DateFormatter
+     */
+    protected $dateFormatter;
+
+    /**
+     * The key-value store service.
+     *
+     * @var \Drupal\Core\KeyValueStore\KeyValueStoreInterface
+     */
+    protected $keyValue;
+
+    /**
+     * The command progress bar service.
+     *
+     * @var \Drush\Drupal\Migrate\MigrateCommandProgressBar
+     */
+    protected $commandProgress;
+
+    /**
+     * Migrate message service.
+     *
+     * @var \Drupal\migrate\MigrateMessageInterface
+     */
+    protected $migrateMessage;
+
+    /**
+     * Constructs a new class instance.
+     *
+     * @param \Drupal\Core\Datetime\DateFormatter $dateFormatter
+     *   Date formatter service.
+     * @param \Drupal\Core\KeyValueStore\KeyValueFactoryInterface $keyValueFactory
+     *   The key-value factory service.
+     * @param \Drush\Drupal\Migrate\MigrateCommandProgressBar $commandProgress
+     *   The command progress bar service.
+     */
+    public function __construct(DateFormatter $dateFormatter, KeyValueFactoryInterface $keyValueFactory, MigrateCommandProgressBar $commandProgress)
+    {
+        parent::__construct();
+        $this->dateFormatter = $dateFormatter;
+        $this->keyValue = $keyValueFactory->get('migrate_last_imported');
+        $this->commandProgress = $commandProgress;
+    }
+
+    /**
+     * List all migrations with current status.
+     *
+     * @command migrate:status
+     *
+     * @drupal-dependencies migrate
```

Comment:
what is `@drupal-dependencies`? Maybe its a dupe of @validate-module-enabled?

Reply:
I have no idea how these tags landed here. It's possible that they were copied from an early variant of `migrate_tools`, still not sure. Removed.